### PR TITLE
OF-1644: Allow non-local users to query MAM.

### PR DESCRIPTION
**Pull request by deleolajide**
This pull request actually adds the required support for MAM version 2 namespace (urn:xmpp:mam:2) (OF-1576) and also allows federated users to fetch permitted chat history. At the moment. user dele@wikisuite.net cannot fetch the chat history of open_chat@igniterealtime.org unless user dele.olajide@igniterealtime.org is used instead.

### CODE REVIEW

**Comment by deleolajide** [comment]
This PR also bumps the MAM namespace for OF-1576

**Comment by guusdk** [comment]
Apart for allowing non-local users to query MAM, this PR also adds support for v0.6 of the [MAM XEP](https://xmpp.org/extensions/xep-0313.html), identified by the `urn:xmpp:mam:2` namespace.

One of the additions to this version of the XEP is described in https://xmpp.org/extensions/xep-0313.html#archives_id. Here, a new requirement ("MUST") is defined that relates to [XEP-0359](https://xmpp.org/extensions/xep-0359.html). I don't believe that the current code has support for XEP-0359-defined "Unique and Stable Stanza IDs" at all (I don't even think it's in Openfire, as I can't find an implementation for the `<stanza-id>` element anywhere). It would therefor be incorrect to advertise support.

**Comment by deleolajide** [comment]
Thanks. I will revert the namespace change.

File: plugins/monitoring/src/java/com/reucon/openfire/plugin/archive/xep0313/IQQueryHandler2.java [hunk 1]
```diff
@@ -30,6 +33,7 @@ protected void sendEndQuery(IQ packet, JID from, QueryRequest queryRequest) {
      */
     private void sendAcknowledgementResult(IQ packet, JID from, QueryRequest queryRequest) {
         IQ result = IQ.createResultIQ(packet);
+        result.setFrom(domain);
```
[the remark this answers is not in the document]
Comment (deleolajide) [reply plugins/monitoring/src/java/com/reucon/openfire/plugin/archive/xep0313/IQQueryHandler2.java]: Probably not, but the IQ.createResultIQ(packet) works as expected providing the correct **from** from the provided **to**. I will change the logic.

**Pull request merged**